### PR TITLE
Decouple pnAsyncCore from some protocol and game logic code

### PR DESCRIPTION
**Pull request by dgelessus**
The AsyncSocket code had one specific assumption about the network protocol hardcoded - this fixes that. (I assume the server-side AsyncSocket code needs to care about this, but the client-side code doesn't.)

Also converts a bunch of `LogMsg` calls outside of pnAsyncCore to more appropriate calls. This also moves the log messages from OLD_ASYNC_LOG.log to more useful log files (e. g. VaultClient.log now actually contains all the vault log messages!).

tldr: Eliminates direct dependencies on pnAsyncCore from places that don't need them.

### CODE REVIEW

**Review by Hoikas — APPROVED** [review]

**Review by Hoikas — APPROVED** [review]

**Comment by Hoikas** [comment]
Is there anything else you wanted to do here, @dgelessus

**Comment by dgelessus** [comment]
No, this PR is complete from my end.
